src/test/java/com/github/fridujo/rabbitmq/mock/exchange/ConsistentHashExchangeTests.java

```diff
@@ -69,8 +67,24 @@ void dispatch_respects_queue_weight() {
 
         assertThat(deliveriesByReceiver).containsOnlyKeys(q1, q2, q3);
 
-        assertThat(Long.valueOf(deliveriesByReceiver.get(q1)).doubleValue() / messagesCount).isCloseTo(0.25D, within(0.01));
-        assertThat(Long.valueOf(deliveriesByReceiver.get(q2)).doubleValue() / messagesCount).isCloseTo(0.5D, within(0.01));
-        assertThat(Long.valueOf(deliveriesByReceiver.get(q3)).doubleValue() / messagesCount).isCloseTo(0.25D, within(0.01));
+        assertThat(Long.valueOf(deliveriesByReceiver.get(q1)).doubleValue() / messagesCount).isCloseTo(0.1D, within(0.01));
+        assertThat(Long.valueOf(deliveriesByReceiver.get(q2)).doubleValue() / messagesCount).isCloseTo(0.7D, within(0.01));
+        assertThat(Long.valueOf(deliveriesByReceiver.get(q3)).doubleValue() / messagesCount).isCloseTo(0.2D, within(0.01));
+    }
+
+
+    @Test
+    void binding_with_non_integer_key_throws_exception() {
```

Comment:
This test can be more expressive with a parameterized test

Reply:
I miss read the message on my phone so I've also parameterised the distribution test to use a 25 tests with a random number of receivers and a random number of messages.

Reply:
I am sorry for the extra work, but to be honest I found the distribution test way less readable.

I am okay with parameterizing it, but I think a few cases with hard coded values will be easier to reason about.

I am also okay with reverting it to the first version 😄

Reply:
No problem.  I've reverted the parameterised version of the dispatch_respects_queue_weight and kept the parameterised version of the exception test.